### PR TITLE
better support for schemes by Franklin Templeton RTA (FTAMIL)

### LINKED ISSUE #5: Missing funds in the database
The below funds seem to be missing or has wrong ids:

Missing:
FTI915-XXXXXX Franklin India Ultra Short Bond Fund- Super Inst. Plan-Segregated Portfolio 1 (erstwhile 8.25% Vodafone Idea Ltd-10jul20-Daily Dividend Option) - Daily - IDCW# Payout

Wrong amc_code: expected "036" vs actual "36"
FTI036-XXXXX Franklin India Prima Fund - GROWTH

Couldn't figure out what's wrong here but the ISIN is not showing up in the parsed output
H60-HDFC Low Duration Fund - Regular Plan - Monthly IDCW# (formerly HDFC Cash Management Fund - Treasury Advantage Plan) - Reinvest

### PR DESCRIPTION
**Pull request by codereverser**
- [x] better support for funds by FTAMIL RTA
- [x] fix issues with HDFC funds
- [x] support CAMS migration of FRANKLIN schemes (+ support old CAS pdf files that may still have FTAMIL as RTA)

fixes #5

### CODE REVIEW

File: casparser_isin/mf_isin.py [hunk 1]
```diff
@@ -81,24 +81,30 @@ def scheme_lookup(self, rta: str, scheme_name: str, rta_code: str):
...
-            amc_code = match.group(1)
-            where.append("amc_code = :amc_code")
-            args.update(amc_code=amc_code)
+        args = {"rta": RTA_MAP.get(str(rta).upper(), "")}
+
+        if "hdfc" in scheme_name.lower() and re.search(r"^h\d+$", rta_code, re.I):
+            # Special case for old HDFC funds with scheme codes of format "H\d+"
+            if re.search("re-*invest", scheme_name, re.I):
+                where.append("name LIKE '%reinvest%'")
+            where.append("rta_code like :rta_code")
```
Comment (ganeshs) [inline casparser_isin/mf_isin.py:100]: This change breaks other scenarios. ex:

`"63"	"HDFC TAX SAVER - GROWTH OPTION"	"INF179K01BB8"	"101979"	"EQUITY"	"CAMS"	"H32"	"32"
"7329"	"HDFC TAX SAVER - DIRECT PLAN - GROWTH OPTION"	"INF179K01YS4"	"119060"	"EQUITY"	"CAMS"	"H32T"	"32T"
`

Comment (codereverser) [reply casparser_isin/mf_isin.py:100]: Can you please post the exact string `"{RTA_CODE}-{SCHEME_NAME_DETAILED}"` from the CAS pdf for the breaking schemes?

Comment (ganeshs) [reply casparser_isin/mf_isin.py:100]: H32-HDFC TaxSaver - Regular Plan - Growth
H02-HDFC Flexi Cap Fund - Regular Plan - Growth

Comment (codereverser) [reply casparser_isin/mf_isin.py]: Tweaked the check a little bit. Can you please test this again if you got some time?

Comment (ganeshs) [reply casparser_isin/mf_isin.py]: Seems to work now. One call out on franklin though, the amfi_code for "Franklin India Ultra Short Bond Fund- Super Inst Plan-Segregated Portfolio 1 **** " is missing

https://www.amfiindia.com/spages/NAVOpen.txt

Comment (codereverser) [reply casparser_isin/mf_isin.py]: yeah. I am aware of it. The amfi code is populated based on the scheme ISIN and it looks like the segragated portfolio schemes in the NAV txt file don't have ISINs. 

working on manually mapping the amfi code for the missing schemes.

Comment (codereverser) [reply casparser_isin/mf_isin.py]: Many thanks for the review. I'll add the AMFI codes for missing schemes soon.

**Pull request merged**